File: solutions/fsharp/tisbury-treasure-hunt/TisburyTreasureHunt.fs
module TisburyTreasureHunt

let getCoordinate (line: string * string) : string = snd line

let convertCoordinate (coordinate: string) : int * char = (int coordinate[0..0], coordinate[1])

let compareRecords (azarasData: string * string) (ruisData: string * (int * char) * string) : bool =
    let (_, azarasCoordinate) = azarasData
    let (_, ruisCoordinate, _) = ruisData
    convertCoordinate azarasCoordinate = ruisCoordinate

let createRecord
    (azarasData: string * string)
    (ruisData: string * (int * char) * string)
    : (string * string * string * string) =
    match (azarasData, ruisData) with
    | ((item, azarasCoordinate), (location, _, color)) when compareRecords azarasData ruisData ->
        (azarasCoordinate, location, color, item)
    | _ -> ("", "", "", "")

File: solutions/julia/tisbury-treasure-hunt/tisbury-treasure-hunt.jl
function get_coordinate(line)
    line[2]
end

function convert_coordinate(coordinate)
    Tuple(coordinate)
end

function compare_records(azara_record, rui_record)
    _, azara_coord = azara_record
    _, rui_coord, _ = rui_record
    convert_coordinate(azara_coord) == rui_coord
end

function create_record(azara_record, rui_record)
    if compare_records(azara_record, rui_record)
        treasure, coord = azara_record
        location, _, quadrant = rui_record
        (coord, location, quadrant, treasure)
    else
        ()
    end
end
